docker-compose.yml

```
@@ -0,0 +1,37 @@
+version: '3.9'
+services:
+  api:
+    build: .
+    volumes:
+      - ./:/src/
+    ports:
+      - '3001:3001'
+    environment:
+      MONGO_URI: ${MONGO_URI}
+      MONGODB_USERNAME: ${MONGO_USERNAME}
+      MONGODB_PASSWORD: ${MONGO_PASSWORD}
+    command: ['npm', 'start']
+
+  mongodb:
+    image: mongo:latest
+    ports: 
+      - '27017:27017'
+    environment:
+      MONGO_INITDB_ROOT_USERNAME: ${MONGO_USERNAME}
+      MONGO_INITDB_ROOT_PASSWORD: ${MONGO_PASSWORD}
+      MONGO_INITDB_DATABASE: shortit
+  mongo-express:
+    image: mongo-express
+    ports: 
+      - '8081:8081'
+    environment:
+      ME_CONFIG_MONGODB_ADMINUSERNAME: ${MONGO_USERNAME}
+      ME_CONFIG_MONGODB_ADMINPASSWORD: ${MONGO_PASSWORD}
+      ME_CONFIG_MONGODB_SERVER: mongodb
+    restart: always
+    depends_on: 
+      - mongodb
+    volumes:
+      - mongodb_data:/data/db
+volumes:
+  mongodb_data:
```

I don't think you need `mongo-express`. You can replace your docker-compose.yml with this.

```suggestion
version: '3.7'
services:
  mongo:
    image: 'mongo'
    restart: unless-stopped
    ports:
      - ${MONGO_PORT}:27017
    env_file:
      - .env
    environment:
      - MONGO_INITDB_ROOT_USERNAME=${MONGO_USERNAME}
      - MONGO_INITDB_ROOT_PASSWORD=${MONGO_PASSWORD}
    volumes:
      - ./data:/data/db

  server:
    build:
      dockerfile: Dockerfile
      context: .
    depends_on:
      - mongo
    env_file:
      - .env
    expose:
      - ${PORT}
    ports:
      - 8000:8000
    volumes:
      - ./src:/app/src
      - ./nodemon.json:/app/nodemon.json
```

Update your env to this.

```
MONGO_USERNAME=admin
MONGO_PASSWORD=pass
MONGO_CLUSTER_NAME=shortit
MONGO_PORT=27017
MONGO_URL=mongodb://${MONGO_USERNAME}:${MONGO_PASSWORD}@mongo:${MONGO_PORT}/${MONGO_CLUSTER_NAME}?authSource=admin
```